Add router nokia-g240wb

elulcao opened the pull request:
- **What does this PR aim to accomplish?:**

Added Router Nokia G-240W-B guide 

- **How does this PR accomplish the above?:**

Added Router Nokia G-240W-B guide for IPv4 and IPv6 addresses 

- **What documentation changes (if any) are needed to support this PR?:**

Added docs/routers/nokia-G240WB.md

![Screen Shot 2022-10-12 at 1 10 44 PM](https://user-images.githubusercontent.com/6373960/195417063-e6d34423-3d82-4bd7-b630-3c8708190a52.png)

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [ x ] I have read the above and my PR is ready for review. _Check this box to confirm_

yubiuser commented on .gitignore:
```diff
@@ -4,3 +4,4 @@
 /.idea
 /node_modules/
 /site/
+Pipfile
```
Why did you add this here?

elulcao replied:
It was used for my local Python env but not intended to merge, to be honest I forgot remove after my last commit. Removed from current PR.

elulcao replied:
Images compressed and resized to a smallest size.

yubiuser reviewed (commented):
Thanks for your submission. Before reviewing your PR in detail, could you please crop/compress+crop your images?

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,112 @@
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+- `Pi-hole` has enabled `DHCP server`.
```
Why do you need Pi-hole to be the DHCP server?

elulcao replied:
Good question, actually this is not necessary since the Router can serve as DHCP server by default.
Removed from guide.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,112 @@
+- `Pi-hole` has enabled `DHCP server`.
+- `Pi-hole` has enabled `IPv6 support (SLAAC + RA)`.
+
+For this example, we will be using the `192.168.1.100` and `2806:103e:1f:face::1681:100` addresses for the following steps. You will need to substitute your own addresses in the following steps.
```
Why do you recommend a global address here? I can change if your ISP assigned a different prefix We usually recommend a ULA address.

https://discourse.pi-hole.net/t/use-ipv6-ula-addresses-for-pi-hole/2127

elulcao replied:
Since this Router doesn't have a section for ULA (explicitly) I configured with IPV6 SLAAC however I realized after reading your comment that the Router offers ULA if the DHCP6 is enabled.

elulcao replied:
I restarted the router and the IPV6 prefix has been updated, however the communication is over the link-local IP.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,107 @@
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
+
+!!! warning
+    Do not use `127.0.0.1#5335` or `::1#5335` as `DNS server` in the `Raspberry Pi` because it will cause a `DNS loop` as the `Pi-hole` will send the queries to the router which in turn will send them to `Pi-hole`.
```
What are you talking about here? Do you have another DNS server running at `127.0.0.1#5335`? Why should this cause a DNS loop?

elulcao replied:
This part is referring to an old configuration in my Raspi, where I was using unbound. I forgot to uninstall unbound and its configurations and hence I was understanding that somehow the Raspi was looping the request due to the `127.0.0.1#5335`. I started again the installation and this time the local IP#5335 is not an issue.

elulcao replied:
Removed the phrase. 
Also I renamed `wlan0` to `eth0` to make the guide homogeneous with current router guides.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+
+Go to **Network** --> **LAN**
+
+1. Set `IPv4 address` to `192.168.1.254`
```
I think there is no need to change the IPv4 address of the router. It might be `192.168.1.1` already by default.

elulcao replied:
It depends of the ISP configuration, in my case 192.168.1.254 is the default GW. I was told that other routers (provided by same ISP are using 192.168.1.1). 
I mentioned that user can set 192.168.1.1 or 192.168.1.254 depending in their default configuration.

elulcao replied:
I configured the router to set static IP to the Raspi, the screenshot isn't including that section; please take a look to the pic attached to this message.

![IMG_1275](https://user-images.githubusercontent.com/6373960/197361022-3315eb37-53b6-466e-8871-6d891588078f.PNG)
* Im using wiregard so I can connect to home lan via LTE.

yubiuser replied:
This is a valid configuration. I think it's a bit cumbersome to set it static on the device and set a static DHCP assignment on the router, but your mileage may vary.  
However, for this public guide I would not recommend others to do so, because it's an additional step they need to perform. It's easier to separate static and dynamic IP pools.

elulcao replied:
I updated the IP address to x.x.1.10 for IPV4, with this change the IP address matches the range used by the DHCP IPV4.

elulcao replied:
Marked as "resolved", please let me know if this is OK to you. :)

yubiuser reviewed (commented):
Looking at your screenshots, your DHCP pool ranges from `.50` - `.250`. However, your Pi's static address `.100` is within the pool. Your router won't know this address is already taken and can hand out this address to some other device.
It's better to give devices static addresses outside of the DHCP pool to avoid this conflict. You should set your Pi's address to something <`.50`

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+
+Go to **Network** --> **LAN**
+
+1. Set `IPv4 address` to the default gateway address, `192.168.1.1` or `192.168.1.254` dependinding on your specific router configuration.
```
```suggestion
1. Set `IPv4 address` to the default gateway address, `192.168.1.1` or `192.168.1.254` depending on your specific router configuration.
```

elulcao replied:
That's a horrible typo!
The word is now correct: `depending`
The pics has been updated.

yubiuser reviewed (commented):
Thanks. Now the screenshots don't fit the new IP of `192.168.1.10` anymore.

elulcao commented:
> Thanks. Now the screenshots don't fit the new IP of `192.168.1.10` anymore.

Updated the pics, thanks for the feedback:)

DL6ER commented:
@yubiuser You reviewed this before, do you still have any objections?

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+This guide was developed using a `Nokia G-240W-B` router. It should work for other models in the `G` series.
```
```suggestion
This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
```

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+This guide was developed using a `Nokia G-240W-B` router. It should work for other models in the `G` series.
+
+### Device information
```
This whole 'Device information' section can go. There is a lot of redundant and/or unnecessary information.

elulcao replied:
Got it, make sense if removing dup info.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+- Software version: `3FE47969AGCA21`
+- Chipset: `BCM6838`
+
+> Note:
```
```suggestion
!!! note
```

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+- Chipset: `BCM6838`
+
+> Note:
+There is no single way to do it right. Chose the one best suited for you.
```
```suggestion
    There is no single way to do it right. Chose the one best suited for you.
```

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively, this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
```
This part should go into the section below (Grab your IPv4 and IPv6 address from your Raspberry Pi).

elulcao replied:
Got you, updated the lines to the correct section.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,104 @@
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
+- The `ISP` provides `IPv6` address, prefix and gateway to the router.
+
+For this example, we will be using the `192.168.1.10` and `fe80::192:168:1:10` addresses for the following steps. You will need to substitute your own addresses in the following steps.
```
I think this part can be deleted. If users set static addresses, they should know them. If not, you provide a section below how to find them.

elulcao replied:
Sure thing, non used info must be removed; including the redundant text.

elulcao commented:
@yubiuser thanks for the review, the guide now has the latest suggestion.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,92 @@
+
+### Prerequisites
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
```
Why did you remove the information about a static IP (or fixed DHCP assignment)?

elulcao replied:
I dont recall why, maybe I misunderstood your last comment. Adding back that section.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,92 @@
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively, this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.
```
I don't understand the second part of this sentence. What do you want to say here

`.... this is intentional to recognize the addresses were manually assigned in the `Raspberry Pi`.`

elulcao replied:
Removing that part since I don't want to confuse users.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,92 @@
+
+- The `Raspberry Pi` is accessible from the `LAN` network via `SSH`.
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
```
Please re-order the paragraphs in this section. It makes sense to start with "SSH into...." followed by the expected output and then everything else.

elulcao replied:
Re-order the text, it must be visible in the next commit.

elulcao commented:
@yubiuser I addressed the last moments, this time the text should be better. Let me know if you want me to do another update to the Readme file. 
Thanks.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,94 @@
+This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
+
+!!! note
+    There is no single way to do it right. Chose the one best suited for you.
```
```suggestion
    There is no single way to do it right. Choose the one best suited for you.
```

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,94 @@
+
+### Prerequisites
+
+- `SSH` into your `Raspberry Pi` and get the `IPv4` and `IPv6` address with `ifconfig eth0` (substituting `eth0` for the applicable network interface).
```
```suggestion

```

This is written blow in the section "Grab yozr IPv4...." already

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,94 @@
+### Prerequisites
+
+- `SSH` into your `Raspberry Pi` and get the `IPv4` and `IPv6` address with `ifconfig eth0` (substituting `eth0` for the applicable network interface).
+- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well.
```
```suggestion
- The `Raspberry Pi` is using static IP address, `IPv4` and `IPv6` as well. This can be a fixed DHCP assignment or configured statically on the Pi itself.
```

elulcao commented:
Hi @yubiuser thanks for the latest comments, I didn't notice the typo. The duplicate entry, I noticed the same but I was unsure if remove or not since it was chained to our previous conversations

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,93 @@
+
+### Grab your IPv4 and IPv6 address from your Raspberry Pi
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively.
```
```suggestion
```

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,93 @@
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
+
+SSH into your `Raspberry Pi`. A quick `ifconfig eth0` (substituting `eth0` for the applicable network interface) will get you the address information:
```
I would put this section and the terminal output as first paragraph in "Grab your IPv4 and IPv6 address from your Raspberry Pi"

elulcao replied:
Addressed the comments for this section.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,93 @@
+
+Notice that the `IPv4` and `IPv6` addresses suffixes matches the number `xxx.168.1.10` and `xxxx::192:168:1:10` respectively.
+
+Worth to mention that the `IPv6` address that starts with `fe80` is the `link-local` address and it is used in this guide, the other addresses that start with `2806` are the global addresses and they are assigned by the router.
```
This could be the last paragraph in "Grab your IPv4 and IPv6 address from your Raspberry Pi"

yubiuser commented on docs/routers/nokia-G240WB.md:
````diff
@@ -0,0 +1,93 @@
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+e.g. `inet 192.168.1.10  netmask 255.255.255.0` and `inet6 fe80::192:168:1:10 prefixlen 64`
````
```suggestion
The relevant lines here are `inet 192.168.1.10  netmask 255.255.255.0` and `inet6 fe80::192:168:1:10 prefixlen 64`
```

yubiuser reviewed (commented):
As (hopefully) last remark: I would re-order the structure of "Grab your IPv4 and IPv6 address from your Raspberry Pi"

And please squash down all your commits into a singel one.

elulcao commented:
> As (hopefully) last remark: I would re-order the structure of "Grab your IPv4 and IPv6 address from your Raspberry Pi"
> 
> And please squash down all your commits into a singel one.

Hello @yubiuser, hope your doing good!
I finally squashed and addressed the comments for this PR 👍 Could you share an +1?
Let me know if anything else is needed.

yubiuser commented on docs/routers/nokia-G240WB.md:
```diff
@@ -0,0 +1,91 @@
+This guide was developed using a `Nokia G-240W-B` router (software version 3FE47969AGCA21). It should work for other models in the `G` series.
+
+!!! note
+There is no single way to do it right. Choose the one best suited for you.
```
I think here is some indention missing, as the 'note' is only an empty box. See the preview: https://deploy-preview-791--pihole-docs.netlify.app/routers/nokia-G240WB/
